src/class-block-converter.php

```diff
@@ -315,21 +318,30 @@ public function remove_image_args( $url ): string {
 		// Split url.
 		$url_parts = wp_parse_url( $url );
 
-		return $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'];
+		if ( isset( $url_parts['query'] ) ) {
+			$url = str_replace( '?' . $url_parts['query'], '', $url );
+		}
+		if ( isset( $url_parts['fragment'] ) ) {
+			$url = str_replace( '#' . $url_parts['fragment'], '', $url );
+		}
+		if ( str_starts_with( $url, '//' ) ) {
+			$url = preg_replace( '#^//#', 'https://', $url );
+		}
+		return $url;
 	}
 
 	/**
 	 * Upload image.
 	 *
 	 * @param string $src Image url.
 	 * @param string $alt Image alt.
-	 * @return string
+	 * @return int|WP_Error
```

Comment:
Should still return `string`. `create_or_get_attachment_from_url` should `throw` if it fails.

Reply:
Updated.